pandas/tests/indexes/period/test_indexing.py

```diff
@@ -685,3 +685,44 @@ def test_period_index_indexer(self):
         tm.assert_frame_equal(df, df.loc[list(idx)])
         tm.assert_frame_equal(df.iloc[0:5], df.loc[idx[0:5]])
         tm.assert_frame_equal(df, df.loc[list(idx)])
+
+    @pytest.mark.parametrize(
+        "key, err ,msg",
+        [
+            (
+                PeriodIndex(["2016"], freq="A"),
+                TypeError,
+                (
+                    r"'PeriodIndex\(\['2016'\], dtype='period\[A-DEC\]', "
+                    r"freq='A-DEC'\)'"
+                    r" is an invalid key"
+                ),
+            ),
+            (2015, KeyError, "2015"),
+            ("foo-bar", KeyError, "Cannot interpret 'foo-bar' as period"),
+            ((2015, 2016), TypeError, r"'\(2015, 2016\)' is an invalid key"),
+        ],
+    )
+    def test_get_loc_not_use_engine(self, key, err, msg):
+        # issue 29234
+        idx = PeriodIndex(
+            ["2014", "2015", "2016", "2017", np.datetime64("NaT")], freq="A"
+        )
+        with pytest.raises(err, match=msg):
+            idx.get_loc(key)
+
+    @pytest.mark.parametrize(
+        "key,result",
+        [
+            (Period(2015, freq="A").ordinal, 1),
+            ("2015-01-01", 1),
+            (datetime(2016, 1, 1), 2),
+            (np.datetime64("NaT"), 4),
+        ],
+    )
+    def test_get_loc_not_use_engine2(self, key, result):
```

Comment:
name the other test test_get_loc_not_use_engine_raises or something; the trailing "2" here isnt informative

Reply:
All of these currently pass on master.  are you sure these aren't redundant with existing tests?